diff --git a/bin/lib/getEditorCommand.sh b/bin/lib/getEditorCommand.sh
index ad5c7ef..3a62c70 100755
--- a/bin/lib/getEditorCommand.sh
+++ b/bin/lib/getEditorCommand.sh
@@ -6,8 +6,6 @@
 #   Where <editor> is the name of your text editor (defaults to $EDITOR) and
 #   <command> is the command to perform (open, hsplit, vsplit).

-source "${ZIDE_DIR}/tmp/env"
-
 # Mapping of common editors and their commands
 declare -A commands=(
   ["hx_cd"]="cd"
diff --git a/bin/zide b/bin/zide
index 23f0eae..13bd12e 100755
--- a/bin/zide
+++ b/bin/zide
@@ -106,7 +106,7 @@ fi
 # Optional params
 #
 # Working directory to open in, defaults to pwd
-cwd="$(cd ${1:-$(pwd)} && pwd) "
+cwd="$(cd "${1:-$(pwd)}" && pwd)"
 # Available layouts can be found in the zide layouts/ directory
 layout=${2:-${ZIDE_DEFAULT_LAYOUT:-"default"}}
 # Default session/tab name to the basename of the provided dir if -N flag set
@@ -113,6 +113,6 @@ layout=${2:-${ZIDE_DEFAULT_LAYOUT:-"default"}}
 # Default session/tab name to the basename of the provided dir if -N flag set
 if [[ "${name}" == "__basename__" ]]; then
-  name="$(basename ${cwd})"
+  name="$(basename "${cwd}")"
 fi

 # Export location of the zide project files for further use
@@ -124,15 +124,19 @@ export ZIDE_ORIGINAL_EDITOR="${EDITOR}"
 export ZIDE_TAB_NAME="${name}"
 export ZIDE_LAYOUT_DIR="${ZIDE_LAYOUT_DIR:-${ZIDE_DIR}/layouts}"

-layout_path="${ZIDE_LAYOUT_DIR}/${layout}"
+layout_path="${ZIDE_LAYOUT_DIR}/${layout%.kdl}.kdl"
 if [ -n "$ZELLIJ" ]; then
-  zellij action new-tab --cwd $cwd --layout "${layout_path}"
+  zellij action new-tab --cwd "$cwd" --layout "${layout_path}"
   zide-rename
 else
   # There's a bug when using zellij options --default-cwd where the CWD isn't working for floating
   # panes defined in the layout, so instead we'll use pushd to change the working dir before
   # starting zellij, and change it back when we exit
-  pushd $cwd > /dev/null
+  pushd "$cwd" > /dev/null
-    zellij --layout "${layout_path}"
+    if [[ -n "${name}" ]]; then
+      zellij attach --create "${name}" options --default-layout "${layout_path}"
+    else
+      zellij --new-session-with-layout "${layout_path}"
+    fi
   popd > /dev/null
 fi
diff --git a/bin/zide-edit b/bin/zide-edit
index 72ab05d..7632c2d 100755
--- a/bin/zide-edit
+++ b/bin/zide-edit
@@ -60,6 +60,7 @@ for path in "${paths[@]}"; do
   escaped_paths+=("$(printf "%q" "$path")")
 done
 path=${escaped_paths[0]}
+raw_path=${paths[0]}

 # Focus editor pane, must be the next pane after yazi
 zellij action focus-next-pane
@@ -68,7 +68,7 @@ zellij action focus-next-pane
 # If we're trying to open a single path and it's a directory, change the working dir to it in our
 # editor. This ensures any other commands in our editor (such as :open, :mv, etc) have the correct
 # cwd set.
-if [ ! -f "${path}" ] && [ ${#escaped_paths[@]} -eq 1 ]; then
+if [ ! -f "${raw_path}" ] && [ ${#escaped_paths[@]} -eq 1 ]; then
   cdCommand="$(getEditorCommand $ZIDE_ORIGINAL_EDITOR cd)"
  zellij action write 27 # send <Escape> key to enter NORMAL mode
  zellij action write-chars ":${cdCommand} ${path}"
diff --git a/layouts/default.kdl b/layouts/default.kdl
index 45f53a8..79bbd7e 100644
--- a/layouts/default.kdl
+++ b/layouts/default.kdl
@@ -1,5 +1,5 @@
 layout {
-    tab {
+    tab hide_floating_panes=true {
         floating_panes {
             zide_rename
         }
@@ -5,7 +5,7 @@ layout {
         }
         pane split_direction="vertical" {
             filepicker size=40 name="picker"
-            editor
+            editor focus=true
         }
         compact_bar size=1
     }
diff --git a/layouts/default_lazygit.kdl b/layouts/default_lazygit.kdl
index 376530e..aa0065d 100644
--- a/layouts/default_lazygit.kdl
+++ b/layouts/default_lazygit.kdl
@@ -6,7 +6,7 @@ layout {
         }
         pane split_direction="vertical" {
             filepicker size=40 name="picker"
-            editor
+            editor focus=true
         }
         compact_bar size=1
     }
